### PR TITLE
Dedicated icons for VFX subgraphs windows

### PR DESCRIPTION
**Pull request by julienamsellem**
### Purpose of this PR
https://fogbugz.unity3d.com/f/cases/1382141/

Any VFX graph window had the same icon, whether it's a simple graph, or a block subgraph or an operator subgraph.
With this PR all three kind of VFX asset will have its own icon displayed in the "tab item" of its window.

Before
subgraphIcons

After
Unity_xXrhbXDljT

---
### Testing status
Tested standard graph, operator subgraph and block subgraph in both light and dark themes.

---
### Comments to reviewers
Notice that if the asset name is too long the window icon is reduced. This is a general behavior in Unity, not specific to VFX Graph editor.

### CODE REVIEW

**Comment by VitaSkruibyte** [comment]
Findings:

- Subgraph Operator icon is barely visible and hard to notice what’s up in the icon. Any chance it is possible to squeeze in the bigger one or it is something against the rules?
- When entering Subgraphs icon does not change
ZxW4KVkjsg

- Is it intended that no asset window has no icon?

Tested:
1. Light theme and dark theme
2. Graphs, subgraph blocks, subgraph operators, no asset windows
3. Import/reimport/refresh
4. Convert to subgraph blocks or operator
5. focus/refocus tab

**Review by VitaSkruibyte — APPROVED** [review]
All the raised concerns were addressed.

Regarding icon being tiny. Per conversation with Julien, icon's size change upon rename. It is also reproducible with other icons (Shader Graph's), for instance.

File: com.unity.visualeffectgraph/Editor/GraphView/VFXViewWindow.cs [hunk 1]
```diff
@@ -16,6 +16,16 @@ namespace UnityEditor.VFX.UI
...
+        private static Dictionary<Tuple<Type, bool>, string> vfxIconMap = new()
+        {
+            { new Tuple<Type, bool>(typeof(VisualEffectSubgraphOperator), true), "d_subgraph-operator.png" },
+            { new Tuple<Type, bool>(typeof(VisualEffectSubgraphBlock), true), "d_subgraph-block.png" },
+            { new Tuple<Type, bool>(typeof(VFXGraph), true), "vfx_graph_icon_gray_dark.png" },
+            { new Tuple<Type, bool>(typeof(VisualEffectSubgraphOperator), false), "subgraph-operator.png" },
+            { new Tuple<Type, bool>(typeof(VisualEffectSubgraphBlock), false), "subgraph-block.png" },
+            { new Tuple<Type, bool>(typeof(VFXGraph), false), "vfx_graph_icon_gray_light.png" },
+        };
+
```
Comment (PaulDemeulenaere) [inline com.unity.visualeffectgraph/Editor/GraphView/VFXViewWindow.cs:28]: Anonymous type FTW \o/
```
private static Dictionary<(Type type, bool proSkin), string> vfxIconMap = new()
{
	{ (typeof(VisualEffectSubgraphOperator), true), "d_subgraph-operator.png" },
	{ (typeof(VisualEffectSubgraphBlock), true), "d_subgraph-block.png" },
	{ (typeof(VFXGraph), true), "vfx_graph_icon_gray_dark.png" },
	{ (typeof(VisualEffectSubgraphOperator), false), "subgraph-operator.png" },
	{ (typeof(VisualEffectSubgraphBlock), false), "subgraph-block.png" },
	{ (typeof(VFXGraph), false), "vfx_graph_icon_gray_light.png" },
};
```
But I'm not sure it works like Tuple when it's used as key 🤔 Let me check ⏳

Comment (PaulDemeulenaere) [reply com.unity.visualeffectgraph/Editor/GraphView/VFXViewWindow.cs:28]: Nice, it seems to works as expected (see [here](https://dotnetfiddle.net/4gVftw))
image

**Review by PaulDemeulenaere — APPROVED** [review]
Only a suggestion of usage of anonymous type instead of tuple (see [comment](https://github.com/Unity-Technologies/Graphics/pull/6633/files#r779401390))
Feel free to apply or ignore.
LGTM! 🟢

File: com.unity.visualeffectgraph/Editor/GraphView/VFXViewWindow.cs (hunk 1 above)
Comment (PaulDemeulenaere) [reply com.unity.visualeffectgraph/Editor/GraphView/VFXViewWindow.cs:28]: So, if you want to apply this change, use `Dictionary<(Type type, bool proSkin), string>` (which can be also readonly btw) and remove occurence of `new Tuple<Type, bool>` 🎉

Comment (julienamsellem) [reply com.unity.visualeffectgraph/Editor/GraphView/VFXViewWindow.cs:28]: Nice 👍

**Pull request merged**